File: app/src/main/java/app/qarya/model/models/requests/FollowRequest.java
package app.qarya.model.models.requests;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class FollowRequest {

    public FollowRequest(Integer id, Boolean follow, String key) {
        this.id = id;
        this.follow = follow;
        this.key = key+id;
        this.value = key+id;
    }

    @SerializedName("id")
    @Expose
    private Integer id;

    @SerializedName("follow")
    @Expose
    private Boolean follow;


    @SerializedName("key")
    @Expose
    private String key;
    @SerializedName("value")
    @Expose
    private String value;



    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Boolean getFollow() {
        return follow;
    }

    public void setFollow(Boolean follow) {
        this.follow = follow;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }
}

File: app/src/main/java/app/qarya/model/models/requests/FollowRequest.kt
package app.qarya.model.models.requests

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class FollowRequest(
    @field:Expose @field:SerializedName("id") var id: Int,
    @field:Expose @field:SerializedName("type") var type: Int,
    @field:Expose @field:SerializedName("follow") var follow: Boolean
)


data class FollowResponse(
     @field:Expose @field:SerializedName("id") val id: Int,
     @field:Expose @field:SerializedName("follow") val followed: Boolean,
     @field:Expose @field:SerializedName("tagKey") val tagKey:String
)